lib/BloqadeExpr/src/types.jl

```diff
@@ -536,6 +536,10 @@ YaoAPI.nqubits(::ThreeLevelRydbergConstGates) = 1
 @inline YaoAPI.nqudits(h::RydbergHamiltonian) = nqudits(h.rydberg_term)
 
 
+function Base.:(==)(lhs::DivByTwo{F1},rhs::DivByTwo{F2}) where {F1,F2}
+    return lhs.f == rhs.f
```

Comment:
I don't think this will always work tho, this only works if the function is identical which is not the case most of the time.

Reply:
cna you elaborate the use case of having this comparison?

Reply:
yeah we don't need this, I thought was a bug in the `==` operator for RydbergHamiltonian but it was not, I forgot to get rid of this!

Reply:
Looks like removing this overload broke my test in BloqadeSchema. 

It's only the test that is comparing two RydbergHamiltonian objects.

Reply:
OK, maybe we will need to have it for now, but I think we should be careful when using this comparison in the future as this may not give what you expect

Reply:
I can just write another test that doesn't rely on the comparison operator.